File: MultiLanguageSandbox/src/test/deps/ruby/15.rb


# Determines the maximum number of drop tests needed to find the drop tolerance height of a device from a building with a given height.
# The function takes a single argument, n, which is the height of the building (number of floors).
# It returns the maximum number of tests needed to find the exact floor from which if the device is dropped, it will break.
#
# The problem assumes the worst-case scenario, where the outcome of each test is the least favorable, thus requiring the maximum number of tests.
#
# Example:
# >>> max_drop_test_height(3)
# 2
# >>> max_drop_test_height(10)
# 4


def max_drop_test_height(n)
  k = 0
  while (k * (k + 1)) / 2 < n
    k += 1
  end
  k
end

raise 'Test failed' unless max_drop_test_height(3) == 2
raise 'Test failed' unless max_drop_test_height(10) == 4
raise 'Test failed' unless max_drop_test_height(15) == 5
raise 'Test failed' unless max_drop_test_height(6) == 3
raise 'Test failed' unless max_drop_test_height(100) == 9


puts 'All tests passed!'

File: MultiLanguageSandbox/src/test/deps/perl/15.pl

sub sorted_squares {
    my @nums = @_;
    my @squared = map { $_ * $_ } @nums;
    my @sorted = sort { $a <=> $b } @squared;
    return @sorted;
}


sub check {
my $func = shift;
die "Test failed!" unless join(',', @{$func->([-4, -1, 0, 3, 10])}) eq '0,1,9,16,100';
die "Test failed!" unless join(',', @{$func->([-7, -3, 2, 3, 11])}) eq '4,9,9,49,121';
die "Test failed!" unless join(',', @{$func->([1, 2, 3, 4])}) eq '1,4,9,16';
die "Test failed!" unless join(',', @{$func->([-1, 0, 1])}) eq '0,1,1';
print "All tests passed!\n";
}

check(\&sorted_squares);